CRM/Case/XMLProcessor/Process.php

```diff
@@ -568,6 +572,106 @@ public function createActivity($activityTypeXML, &$params) {
     return TRUE;
   }
 
+  /**
+   * Return the default assignee contact for the activity.
+   *
+   * @param array $activityParams
+   * @param object $activityTypeXML
+   * @return int|null the ID of the default assignee contact or null if none.
+   */
+  protected function getDefaultAssigneeForActivity($activityParams, $activityTypeXML) {
+    if (!isset($activityTypeXML->default_assignee_type)) {
+      return NULL;
+    }
+
+    $this->setupDefaultAssigneeOptionIds();
+
+    switch($activityTypeXML->default_assignee_type) {
+      case $this->defaultAssigneeOptionsIds['BY_RELATIONSHIP']:
+        return $this->getDefaultAssigneeByRelationship($activityParams, $activityTypeXML);
+        break;
+      case $this->defaultAssigneeOptionsIds['SPECIFIC_CONTACT']:
+        return $this->getDefaultAssigneeBySpecificContact($activityTypeXML);
+        break;
+      case $this->defaultAssigneeOptionsIds['USER_CREATING_THE_CASE']:
+        return $activityParams['source_contact_id'];
+        break;
+      case $this->defaultAssigneeOptionsIds['NONE']:
+      default:
+        return null;
+    }
+  }
+
+  /**
+   * Fetches and caches the activity's default assignee options.
+   *
+   * @return array
+   */
+  protected function setupDefaultAssigneeOptionIds() {
+    if (!empty($this->defaultAssigneeOptionsIds)) {
+      return $defaultAssigneeOptionsIds;
+    }
+
+    $defaultAssigneeOptions =  civicrm_api3('OptionValue', 'get', array(
+      'option_group_id' => 'activity_default_assignee',
+      'options' => array('limit' => 0)
+    ));
+
+    foreach($defaultAssigneeOptions['values'] as $option) {
+      $this->defaultAssigneeOptionsIds[$option['name']] = $option['id'];
+    }
+  }
+
+  /**
+   * Returns the default assignee for the activity by searching for the target's
+   * contact relationship type defined in the activity's details.
+   *
+   * @param array $activityParams
+   * @param object $activityTypeXML
+   * @return int|null the ID of the default assignee contact or null if none.
+   */
+  protected function getDefaultAssigneeByRelationship($activityParams, $activityTypeXML) {
+    if (!isset($activityTypeXML->default_assignee_relationship)) {
+      return NULL;
+    }
+
+    $targetContactId = is_array($activityParams['target_contact_id'])
+      ? $activityParams['target_contact_id'][0]
+      : $activityParams['target_contact_id'];
+
+    $relationships = civicrm_api3('Relationship', 'get', array(
+      'contact_id_b' => $targetContactId,
+      'relationship_type_id.name' => $activityTypeXML->default_assignee_relationship,
+      'sequential' => 1
+    ));
```

Comment:
I'm not sure about the requirements, but maybe we should only consider active relationships here? An active relationship has the `is_active` flag set to true and the start and end date (if not null), containing the current date.